Sources/Subs-Db-postgresql.php

```diff
@@ -707,7 +707,7 @@ function smf_db_insert($method = 'replace', $table, $columns, $data, $keys, $ret
 	$table = str_replace('{db_prefix}', $db_prefix, $table);
 
 	// Sanity check for replace is key part of the columns array
-	if ($method == 'replace' && count(array_intersect_key($columns, array_flip($keys))) !== count($keys))
+	if ($method == 'replace' && (empty($keys) || count(array_intersect_key($columns, array_flip($keys))) !== count($keys)))
```

Comment:
Looks good but can you separate the empty($keys) into its own check?  just so we can provide a more detailed error message and help whoever uses the replace method to debug their code.

Reply:
done

Reply:
cool, thanks!